crates/engine/tree/src/tree/mod.rs

```diff
@@ -1999,6 +2031,27 @@ where
     }
 }
 
+impl<P: Debug, E: Debug, T: EngineTypes + Debug> std::fmt::Debug for EngineApiTreeHandler<P, E, T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
```

Comment:
if we require that trait InvalidBlockHook : Debug, we can likely avoid that

Reply:
done

Reply:
hmm actually unsure, because then we won't be able to use raw functions as hooks, because `fn()` doesn't implement `Debug`